.github/workflows/cli-verification.yaml

```diff
@@ -0,0 +1,42 @@
+name: CLI Publish & Verify
+
+on:
+    push:
+        branches:
+            - master
+            - staging/**
+    pull_request:
+
+concurrency:
+    group: verify-cli-${{ github.event.pull_request.number || github.ref }}
+    cancel-in-progress: true
+
+jobs:
+    publish-and-test:
+        runs-on: ubuntu-latest
+        permissions:
+            contents: read
+            packages: write
+        steps:
+            - uses: actions/checkout@v4
+
+            - uses: actions/setup-node@v4
+              with:
+                  node-version: '18.x'
+                  registry-url: 'https://npm.pkg.github.com'
+                  scope: '@nangohq'
+                  always-auth: true
+            - name: Publish npm packages to the github registry
+              env:
+                  NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+              shell: bash
+              run: |
+                  GIT_HASH=$(git rev-parse HEAD)
+                  bash ./scripts/publish.sh 0.0.1-$GIT_HASH
+            - name: Install the cli from the github package registry
+              env:
+                  NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+              run: |
+                  GIT_HASH=$(git rev-parse HEAD)
+                  npm install nango@0.0.1-$GIT_HASH
+                  npx nango version --debug
```

Comment:
If the goal is to test the cli I think I would prefer we don't publish each commit to npm, 
`build + npm install -g packages/cli` should work no?

Reply:
Thanks. The goal is to check the packaging of the cli package to a remote source -- in this case github registry. By packaging I mean the changes in https://github.com/NangoHQ/nango/pull/1918

If I install the cli via `npm install -g packages/cli` I don't think that'll mimic the same process as I would like to test that npm is doing when we upload a packed dependency.

Reply:
Added more info to the PR description